docs/DeveloperGuide.md

```diff
@@ -270,58 +270,208 @@ _{Explain here how the data archiving feature will be implemented}_
+
+
+**3. Use case: UC03 - Marking Project Status**
+
+**Guarantees:**
+* Project Status goes from “Yet to Start” (YTS) to “In Progress”  (IP); (or alternatively, from “In Progress” (IP) to “Done” (DONE).
+
+
+**Preconditions: Existence of said FYP in database**
+
+**MSS**
+
+1.  Professor wants to modify the status of his student’s project, and he/she specifies the desired status change.
+2.  JerylFypManager toggles the status of the project.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. Professor’s input of Student’s ID / Status Tag is incorrect (e.g. misspell) / accessing FYP that does not exist.
+    * 1a1. JerylFypManager shows an error message.
+    * 1a2. Professor keys in his input again.
+
+    Steps 1a1-1a2 are repeated until the request entered is correct.
 
-**Use case: Delete a person**
+  Use case resumes from step 2.
+
+
+**4. Use case: UC04 - Removing FYP**
+
+**Preconditions: Professor can recall the name/id of the FYP**
+
+**MSS**
+
+1.  Professor requests to remove a student FYP into the list.
+2.  JerylFypManager removes the FYP details.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The request given is invalid.
+    * 1a1. JerylFypManager shows an error message.
+    * 1a2. Professor enters a new request.
+
+  Steps 1a1-1a2 are repeated until the request entered is correct.
+
+  Use case resumes from step 2.
+
+
+**5. Use case: UC05 - Help List**
+
+**Preconditions: Nil**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  Professor requests from JerylFypManager to list all possible commands.
+2.  JerylFypManager shows the list of possible commands and format to be entered in.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* Nil
+
+
+**6. Use case: UC06 - List FYPs**
+
+**Preconditions: There exist FYP(s) in JerylFypManager**
+
+**MSS**
+
+1.  Professor requests for JerylFypManager to show the list of FYPs.
+2.  JerylFypManager shows a list of FYPs to the professor.
+
+    Use case ends.
+
+
+**Extensions**
+
+* 2a. There are currently no FYPs that the professor is overseeing.
+    * 2a1. JerylFypManager tells the professor that he is currently not overseeing any FYPs.
 
   Use case ends.
 
-* 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+**7. Use case: UC07 - Exit JerylFypManager**
+
+**Preconditions: Nil**
+
+**MSS**
+
+1.  Professor requests to exit JerylFypManager.
+2.  JerylFypManager shows exit message.
+3.  JerylFypManager application is closed.
+
+    Use case ends.
+
+**Extensions**
+
+* Nil
 
-      Use case resumes at step 2.
 
 *{More to be added}*
 
+--------------------------------------------------------------------------------------------------------------------
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
```

Comment:
Add some NFT here. Do check if they align with our project and what other NFT we should consider

Reply:
*NFR 😆